storage/keyring.go

```diff
@@ -165,34 +168,124 @@ var storageDataUnmarshal Unmarshaler = json.Unmarshal
 
 // loads the entire StorageData into memory
 func (kr *KeyringStore) getStorageData(s *StorageData) error {
-	data, err := kr.keyring.Get(RECORD_KEY)
+	var data []byte
+	var err error
+
+	switch kr.getMyGOOS() {
+	case "windows":
+		data, err = kr.joinAndGetKeyringData(RECORD_KEY)
+	default:
+		data, err = kr.getKeyringData(RECORD_KEY)
+	}
+
 	if err != nil {
 		// Didn't find anything in our keyring
 		*s = NewStorageData()
 		return nil
 	}
-	if err = storageDataUnmarshal(data.Data, s); err != nil {
+	if err = storageDataUnmarshal(data, s); err != nil {
 		return err
 	}
 	return nil
 }
 
+// reads a single entry of the keyring
+func (kr *KeyringStore) getKeyringData(key string) ([]byte, error) {
+	data, err := kr.keyring.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	return data.Data, nil
+}
+
+// read the chunks stored in windows credential manager
+func (kr *KeyringStore) joinAndGetKeyringData(key string) ([]byte, error) {
+	i := 0
+	data := []byte{}
+	var err error
+	var payload_size int
+	var chunk []byte
+
+	chunk, err = kr.getKeyringData(fmt.Sprintf("%s_%d", key, i))
+	if err != nil {
+		return nil, err
+	}
+	if len(chunk) < 8 {
+		return nil, fmt.Errorf("Invalid stored data in Keyring. There's not enough data")
+	}
+
+	payload_size, chunk = int(getUINT64FromBytes(chunk[:8])), chunk[8:]
```

Comment:
the length is stored as a _unsigned 64_ bit integer, but it's being converted to a signed 64 or 32 bit integer (platform dependent)?   wasn't `payload_size` a uint64 before?

while unlikely, this would render the data unreadable should it exceed 2^31-1 on certain platforms.

Reply:
We need a signed because if you modify a chunk i.e the number 2 with more data, when you do the difference you'll get an integer overflow.

Reply:
So normally I would block a merge for this because it really is doable, but sounds like you're not sure how to do that and frankly you've already dramatically improved this PR since it was initially submitted so I'm happy to clean it up in post for you.

Reply:
Feel free to block the PR if you aren't comfortable with my code. As I said in my first post I don't develop usually,... and less code production-ready. 

I think last time I dit, I was sending you a patch to make this project work on windows  xD

I'm learning a lot with this PR and your comments, so I'm happy with the result. of this code-review.